packages/expo-router/src/fork/getStateFromPath.ts

```diff
@@ -671,9 +708,9 @@ const createConfigItem = (
     path,
     // The routeNames array is mutated, so copy it to keep the current state
     routeNames: [...routeNames],
-    expandedRouteNames: screen.includes('/')
-      ? [...routeNames.slice(0, -1), ...screen.split('/')]
-      : [...routeNames],
+    expandedRouteNames: routeNames.flatMap((name) => {
+      return name.split('/');
```

Comment:
note: I can't immediately tell from the code whether the normalization from before is redundant